### PR TITLE
Remote connection status

### LINKED ISSUE #9859: Remote Content Browsing: Update UI to display connection errors
## Overview 

This issue is a full-stack issue to ensure that the connection states are properly displayed in the content renderer and throughout the experience of browsing other remote content libraries

This implementation depends on the `useDevices` composable which supports remote content browsing.

### Figma 

Screenshot 2023-02-13 at 5 32 48 PM

Screenshot 2023-02-13 at 5 33 12 PM

Screenshot 2023-02-13 at 5 33 40 PM

## Acceptance Criteria
- [ ] On the user's library main page - connection status to other devices and the network is correctly displayed, with connection, disconnection, and searching states
- [ ] Error handling is implemented which prioritizes persistence of the display of the current page whenever possible, and prioritizes managing a disconnected state message over our general error state "something went wrong" if at all possible
     - [ ] When browsing remote content, if the connection drops, the disconnection indicator appears in the right corner
     - [ ] The user receives a snackbar notification that the connection has dropped

## References 
- [Notion project page](https://www.notion.so/learningequality/Remote-Content-Syncing-Browse-Libraries-dab6e6d520d642bb80ed68f3b37e93fa)
- [Spec doc](https://www.notion.so/learningequality/Remote-Content-Browsing-Reference-Spec-s-500ab68669cb4b1fb2e752314d21f6cb)
- [Figma spec](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=194%3A2922&t=UZhtEmvHSEWEKomz-0)
- [Gherkin scenarios](https://www.notion.so/learningequality/Gherkin-scenarios-02096ac9925844f8902c818cab607d4b#0a205e75b2ea49a9be07acf15bf13b7a)

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary
Adds a connection status icon when browsing a remote library content
Peek 2023-03-29 20-13

## References
Closes: #9859 

## Reviewer guidance
As in the attached video, explore a remote library and connect and disconnect the remote device to test if it works fine.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by marcellamaki — COMMENTED** [review]
This looks good to me and initial manual QA checks out. @radinamatic can @pcenov confirm this Monday morning his time?

File: kolibri/plugins/learn/assets/src/views/LearningActivityBar.vue [hunk 1]
```diff
@@ -66,6 +66,8 @@
             </div>
           </template>
         </CoreMenu>
+        <DeviceConnectionStatus :deviceId="deviceId" />
```
Comment (akolson) [inline kolibri/plugins/learn/assets/src/views/LearningActivityBar.vue:69]: Non-blocking, but just double checking you intentionally left out the color prop here :)

**Review by akolson — APPROVED** [review]
Thanks @jredrejo. This looks good and the implementation is very clear.

**Comment by pcenov** [comment]
Hi @jredrejo, I have some questions because while I am able to see the 'Disconnected' status icon as shown in your example I am also observing the following issues and it's not clear to me whether these are in scope of this PR or are separate issues:

1. Why is the 'Disconnected' status icon being displayed for channels and resources which are on my own device?
2. When I click on a channel on the remote device I can't see the channel's resources and therefore I can't test whether the 'Disconnected' status icon is being displayed there while viewing the resources as specified in https://github.com/learningequality/kolibri/issues/9859. The following error is displayed in the console: `TypeError: Cannot read properties of undefined (reading 'page')`
3. When I refresh the 'Explore libraries' page for a moment I see the 'Disconnected' status icon.

Video:

https://user-images.githubusercontent.com/79847249/229489520-a5739d98-689f-4052-98b2-88c7b79c73d9.mp4

**Pull request merged**